hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -602,6 +603,10 @@ void Vgen::emit(const andli& i) {
   a->And(W(i.d), W(i.s1), MSKTOP(i.s0.l()), SetFlags);
 }
 
+void Vgen::emit(const testbi& i) {
+  a->Tst(W(i.s1), MSKTOP(i.s0.l()));
+}
```

Comment:
I think we _should_ use `.b()` here, and that we still need a new MSKTOP24. Using `b()` will ensure that the new value can be cast as a byte, but we still need to get rid of the upper 24 bits before testing it. `tst` is an alias of `ands`. We are using W registers here, but if there are sign bits in that second byte of the immediate, then `ands` will use them.

I think as it coded now, the implementation of `testbi` is incorrect.